ui/app/pages/create-account/connect-hardware/index.js

```diff
@@ -159,18 +160,28 @@ class ConnectHardwareForm extends Component {
       });
   };
 
-  onUnlockAccount = (device) => {
+  onUnlockAccounts = (device, path) => {
     const {
       history,
       mostRecentOverviewPage,
-      unlockHardwareWalletAccount,
+      unlockHardwareWalletAccounts,
     } = this.props;
+    const { selectedAccounts } = this.state;
 
-    if (this.state.selectedAccount === null) {
+    if (selectedAccounts.length === 0) {
       this.setState({ error: this.context.t('accountSelectionRequired') });
     }
 
-    unlockHardwareWalletAccount(this.state.selectedAccount, device)
```

Comment:
Curious that `hdPath` wasn't being set before :thinking: 
I investigated this and found that the HD path was being set correctly when we retrieve a page of accounts, and it persists in the keyring after that point. So luckily it's _almost_ always set correctly.

I did discover that on `develop` if you add an account on one path, start the connect flow again, switch paths, then cancel, it leaves the pre-existing account but in a broken state. So that's interesting. That is fixed by this PR though.